astropy/coordinates/tests/test_angles.py

```diff
@@ -1092,3 +1092,33 @@ def test_str_repr_angles_nan(cls, input, expstr, exprepr):
     # Deleting whitespaces since repr appears to be adding them for some values
     # making the test fail.
     assert repr(q).replace(" ", "") == f'<{cls.__name__}{exprepr}>'.replace(" ","")
+
+
+@pytest.mark.parametrize(
+    "value,expected_value,dtype",
+    [
+        (np.pi / 2, np.pi / 2, np.float64),
+        (np.float32(np.pi / 2), np.float32(np.pi / 2), np.float32),
+        (np.float32(np.pi / 2), np.pi / 2, np.float64),
+        (np.float32(np.pi / 2), np.pi / 2, None),
```

Comment:
```suggestion
        (np.float32(np.pi / 2), np.float(np.pi / 2), None),
```
This will (and does successfully) create an `Angle` of the input dtype, so you need to compare against that.